chore(tools): harmonize type and scope git guidelines (LIVE-27608) (#19017)
LedgerHQ/ledger-live: CONTRIBUTING.md

### Git Conventions

We use the following git conventions for the `ledger-live` monorepo. These align with the [ADR: Harmonize type and scope](https://ledgerhq.atlassian.net/wiki/spaces/WXP/pages/7266992139/ADR+Harmonize+type+and+scope+across+pull+request+titles+and+GitHub+labels).

#### Scope values

> Scope identifies which part of the monorepo a change affects. The canonical list lives in [`.github/scopes.ts`](.github/scopes.ts).

Author: **suggestion: add note about link to labels**

```suggestion
Scope identifies which part of the monorepo a change affects. The canonical list lives in [`.github/scopes.ts`](.github/scopes.ts). It is also used by GitHub to apply labels to pull requests.
```

Examples: `desktop`, `mobile`, `common`, `coin-modules`, `shared-lib`, `ui`, `translations`.

Scope values must be lowercase and use hyphens for multi-word names (for example `coin-modules`, `shared-lib`).

> 

Author: **suggestion: remove unnecessary extra detail**

```suggestion
```

[`.github/labeler.yml`](.github/labeler.yml) is generated from [`.github/scopes.ts`](.github/scopes.ts). After changing scopes or paths, run `pnpm generate:labeler`.

> 

Author: **suggestion: move notes about GitHub labels to lines above**

```suggestion
```

#### Commit types

Type describes the nature of a change. Use one of the following values (aligned with [Conventional Commits](https://www.conventionalcommits.org/) and enforced by [commitlint](commitlint.config.js)):

| Type | Use for |
| --- | --- |
| `feat` | New feature |
| `fix` | Bug fix |
| `docs` | Documentation only |
| `style` | Formatting, no code change |
| `refactor` | Restructure without behavior change |
| `perf` | Performance improvement |
| `test` | Add or update tests |
| `build` | Build system or external dependencies |
| `ci` | CI/CD configuration |
| `chore` | Maintenance, tooling, configs |
| `revert` | Revert a previous commit |

#### Branch naming

Branch names are recommended but not enforced. Use this pattern:

```

> <type>/<scope>-<ticket>-<short-description>

Author: **suggestion: make scope and ticket optional**

```suggestion
<type>/[<scope>][-<ticket>]-<short-description>
```

```

> - **type** — one of the [commit types](#commit-types) above (for example `feat`, `fix`, `chore`)
> - **scope** — a [scope](#scope-values) (for example `desktop`, `mobile`, `common`)
> - **ticket** — Jira issue key when applicable (for example `LIVE-27608`); omit for external contributors without a ticket
> - **short-description** — kebab-case summary of the change

Author: **suggestion: make scope and ticket optional**

```suggestion
- **type** — one of the [commit types](#commit-types) above (for example `feat`, `fix`, `chore`)
- **scope** — a [scope](#scope-values) (for example `desktop`, `mobile`, `common`) (optional)
- **ticket** — Jira issue key when applicable (for example `LIVE-27608`); (optional)
- **short-description** — kebab-case summary of the change
```

- **gitmoji** — not used in this repository

You can use the `pnpm commit` prompt to ensure that your commit messages are valid, as well as the `pnpm commitlint --from <target branch>` command to check that every commit on your current branch are valid.

Examples:

```
feat(desktop): add dark mode toggle
fix(mobile): resolve transaction signing issue
docs(common): update API documentation
refactor(coin-modules): simplify account syncing logic
test(shared-lib): add client-ids unit tests
```

#### Pull request title

Pull request titles follow the same `type(scope): description` pattern as [commit messages](#commit-message).

Examples:

```
feat(desktop): add dark mode toggle
fix(coin-modules): correct fee estimation for bitcoin
chore(automation): harmonize scope definitions
```

A GitHub workflow may automatically prepend a scope prefix based on changed files:

- `[LWD]` — Desktop-only changes
- `[LWM]` — Mobile-only changes
- `[LWDM]` — Common, shared, or cross-platform changes (including both Desktop and Mobile)

The resulting title looks like `[LWD] feat(desktop): add dark mode toggle`. This prefix is used by Jira fix-version automation and should be left in place.

> PR titles should be reusable for merge commit naming when squashing or rebasing.

Author: ```suggestion
PR titles should be reusable for merge commit naming.
```

### Git Conventions

> We use the following git conventions for the `ledger-live` monorepo. These align with the [ADR: Harmonize type and scope](https://ledgerhq.atlassian.net/wiki/spaces/WXP/pages/7266992139/ADR+Harmonize+type+and+scope+across+pull+request+titles+and+GitHub+labels).

Author: **suggestion: Remove noise**

```suggestion
We follow conventions for commit messages, pull request titles and branch names. The conventions include consistent naming of the "scope" and "type" of each change. 
```

Scope values must be lowercase and use hyphens for multi-word names (for example `coin-modules`, `shared-lib`).

[`.github/labeler.yml`](.github/labeler.yml) is generated from scope paths and additional label definitions. After changing [`.github/scopes.ts`](.github/scopes.ts) or [`.github/labeler.yml.ts`](.github/labeler.yml.ts), run `pnpm generate:labeler`.

#### Commit types

Type describes the nature of a change. Use one of the following values (aligned with [Conventional Commits](https://www.conventionalcommits.org/) and enforced by [commitlint](commitlint.config.js)):

| Type | Use for |
| --- | --- |
| `feat` | New feature |
| `fix` | Bug fix |
| `docs` | Documentation only |
| `style` | Formatting, no code change |
| `refactor` | Restructure without behavior change |
| `perf` | Performance improvement |
| `test` | Add or update tests |
| `build` | Build system or external dependencies |
| `ci` | CI/CD configuration |
| `chore` | Maintenance, tooling, configs |
| `revert` | Revert a previous commit |

#### Branch naming

Branch names are recommended but not enforced. Use this pattern:

```

> <type>/<scope>-<ticket>-<short-description>

Author: Good call. I'll carry this suggestion forward into a comment of my own.
